src/libOpenImageIO/color_ocio.cpp

```diff
@@ -49,6 +51,64 @@ namespace OCIO = OCIO_NAMESPACE;
 OIIO_NAMESPACE_BEGIN
 
 
+// Class used as the key to index color processors in the cache.
+class ColorProcCacheKey {
+public:
+    ColorProcCacheKey (ustring in, ustring out,
+                       ustring key=ustring(), ustring val=ustring(),
+                       ustring looks=ustring(), ustring display=ustring(),
+                       ustring view=ustring(), ustring file=ustring(),
+                       bool inverse=false)
+        : inputColorSpace(in), outputColorSpace(out),
+          context_key(key), context_value(val), looks(looks), file(file),
+          inverse(inverse)
+    {
+        hash = (inputColorSpace.hash() ^ outputColorSpace.hash() ^
```

Comment:
I don't think xoring the hash values together is a great way to combine them (not that we have a convenient way around this right now).

Reply:
This is not going to be a heavily contended hash table -- any run is going to have a fairly limited number of transformations (because there are a small number of color space names).

Reply:
Oh, though I see the a particular case is problematic: it will hash from-to and to-from identically. I will add some multipliers in there to prevent that.